components/heartbeat/STM32F103C8.ld

```diff
@@ -0,0 +1,139 @@
+/**
+ * STM32F103C8.ld
+ * Linker script for STM32F103C8
+ */
+
+/* Entry Point */
+ENTRY(Reset_Handler)
+
+/* Highest address of the user mode stack */
+_estack = ORIGIN(RAM) + LENGTH(RAM);    /* bottom of stack (grows down) */
+
+__STACK_SIZE = 0x400;
+__HEAP_SIZE = 0x200;
+
+__RAM_SIZE = 20K;
+__FLASH_SIZE = 58K;

```

Comment:
I think we would benefit from allocation a 1kB nvm area at the back of flash. Will be useful to integrate it from the start for all of the bms and other systems

Reply:
once the bootloader size is stabilized we can allocate an extra page at the beginning of the flash for nvm. iirc the page sizes are bigger at the end of flash than at the beginning, so ideally we want to allocate them at the beginning of flash for the bootloader so they don't consume too much application space. The apps can use whatever is left at the end of flash for nvm, unless there's less than a page. If we run into that, we could also explore having the app's nvm pages at the front too

Reply:
I was wrong about this. On this chip, all flash pages are 1k. We can allocate wherever, but probably not worth doing until we actually have a need for nvm